test/integration/Test/Integration/Scenario/Transactions.hs

```
@@ -344,35 +345,42 @@ spec = do
     multioutputTransactionScenario
         :: String
         -> [Word64]
-        -> [Word64]
+        -> Maybe (TwoElementList Word64)
```

This shouldn't be a `Maybe`; this is made fairly obvious by the way the function handles that with a `case` right away, failing hard in the case it receives a `Nothing`. Generally-speaking, make the input domain no more than what you need. 
This is up to the caller to produce a value of the right type.

---

yes, the caller should take responsibility here